File: exercises/ddd_infrastructure_layer/java/src/main/java/domain/application/FullName.java
package domain.application;
public class FullName {
    private final String value;

    public FullName(String value) {
        if (!isValidName(value)) {
            throw new IllegalArgumentException("Invalid full name.");
        }
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    private boolean isValidName(String name) {
        return name != null && !name.trim().isEmpty();
    }
}

File: exercises/ddd_infrastructure_layer/typescript/src/domain/application/FullName.ts
export class FullName {
  private readonly _value: string;

  get value() { return this._value; }

  constructor(value: string) {
    if (!this.isValidName(value)) {
      throw new Error(`Invalid full name: ${value}`);
    }
    this._value = value;
  }

  private isValidName(value: string): boolean {
    return !!value && value.trim().length > 0;
  }
}
